syntax = "proto3";

package lcm_calculator;

message SessionRequest {
    string session_id = 1;
}

message CalculateRequest {
    string session_id = 1;
    repeated string value = 2;
}

message Response {
    enum SessionState {
        OK = 0;
        BROKEN = 2;
        CLOSED = 3;
    }
    string session_id = 1;
    SessionState state = 2;
    string result = 3;
}

service Calculator {
    rpc BeginSession(SessionRequest) returns (Response) {}
    rpc Calculate(CalculateRequest) returns (Response) {}
    rpc EndSession(SessionRequest) returns (Response) {}
}
